R/defStormsDataset.R

```diff
@@ -241,23 +248,350 @@ checkInputsdefStormsDataset <- function(filename, fields, basin, seasons, unitCo
+
+      } else if (unitConversion["poci"] == "psi2pa") {
+        poci[,i] <- as.numeric(c(psi2pa(as.numeric(dataBaseFiltered[start:end, fields["poci"]])), rep(NaN, row-countObs[i])))
+
+      } else if (unitConversion["poci"] == "atm2pa") {
+        poci[,i] <- as.numeric(c(atm2pa(as.numeric(dataBaseFiltered[start:end, fields["poci"]])), rep(NaN, row-countObs[i])))
+
+      } else {
+        poci[,i] <- as.numeric(c(dataBaseFiltered[start:end, fields["poci"]], rep(NaN, row-countObs[i])))
+      }
+    }
+
+  }
+  
+  # Collect data
+  data <- list(
+    names = names,
+    seasons = seasons,
+    isotimes = isotimes,
+    longitude = longitude,
+    latitude = latitude,
+    msw = msw
+  )
+  
+  # TODO Remove later
+  if("sshs" %in% names(fields)){
+    data$sshs <- sshs
+  }
+  
+  if("rmw" %in% names(fields)){
+    data$rmw <- rmw
+  }
+  
+  if("pressure" %in% names(fields)){
+    data$pressure <- pressure
+  }
+  
+  if("poci" %in% names(fields)){
+    data$poci <- poci
+  }
+  
+  
+  return(data)
+
+}
+
+
+
+
+################
+# Main function#
+################
+
 #' Creating a `stormsDataset` object
 #'
-#' The `defStormsDataset()` function creates a `stormsDataset` object from a NetCDF file.
+#' The `defStormsDataset()` function creates a `stormsDataset` object from either a NetCDF or a CSV file.
 #' This is an essential first step before other `stormR` functions can be used.
 #'
-#' @param filename character. Name of the NetCDF (.nc) file. Default is the `test_dataset.nc`
+#' @param filename character. Name of the NetCDF (.nc)/CSV (.csv) file. Default is the `test_dataset.nc`
 #' file located in the `inst/extdata` repository of the directory (accessible by
 #' `system.file("extdata", "test_dataset.nc", package = "StormR")`). This test dataset is extracted
 #' from the IBTrACS.SP.v04r00.nc file and provides all the tropical cyclones that occurred around Vanuatu
 #' from 2015 to 2016 and around New Caledonia from 2020 to 2021.
-#' @param fields named character vector. This argument allows to specify the corresponding variable names
-#'  in the input NetCDF file for each field in the output `stormsDataset`. By default, the corresponding
-#'  variable names are set up to import data from a NetCDF file from the IBTrACS database (Knapp et al., 2010).
+#' @param fields named character vector. This argument allows to specify the corresponding variable names (NetCDF) or
+#' headers (CSV) in the input file for each field in the output `stormsDataset`. By default, the corresponding
+#' variable names are set up to import data from a NetCDF file from the IBTrACS database (Knapp et al., 2010).
 #' Corresponding variable names for following fields have to (mandatory fields) or can be
 #' (recommended or optional fields) provided:
 #'  \itemize{
-#'    \item "`names"`, names of the storms (mandatory),
+#'    \item `"sid"`, unique identifiers of the storms (mandatory if `filename` is a CSV)
```

Comment:
Why ?
This should be removed

Reply:
Not sure about that, the SID is the only variable that is unique by storm, Imagine a dataset where two storms share both the same names and season, I agree it should be weird but not impossible ?
I need it to make sure to get unique storms (not the case while doing a `unique(names)` for example,

How would you proceed ?

Reply:
For netcdf, this is not a problem because each storm is contained within a single unique slot. In a csv, each row represents an observation  and are stacked for  each storms

Reply:
I cannot handle "NOT_NAMED" storm like so, unless I previously remove them from the observations, So I suggest keeping SID for the moment and think about what we should do with "NOT_NAMED"  and SID later

Reply:
What should we do then ?

Reply:
Yes, I saw that earlier today, we should be consistent with that. What is it is not "NOT_NAMED" but another constant for not named storms in the dataset ? Maybe we should add an input like `notNamed` which is by default "NOT_NAMED" as it is the case for ibrtracs dataset ?

Reply:
I'm not sure to get your question... 
It is very unlikely that `NOT_NAMED` or `notNamed` appears in any csv dataset. This makes me think of 2 things:
- the fact that we hard coded the "NOT_NAMED" removal for netcdf is not good: it is specific to IBTRACKS. We should remove that.
- We must think of a way to remove them in a consistent way for both netcdf and csv. One could be to propose an argument for `defStormsDataset` for not named storms, which by default is `NOT_NAMED`, and remove those storms.

Reply:
Yes that is exactly my point, you just explained it better than me

Reply:
Done